doc/src/devdocs/types.md

```diff
@@ -455,129 +335,34 @@ breakpoints in other functions.
-$4 = 2
-(gdb) call jl_(eqc->data[0])
-＃T<:Any
-(gdb) call jl_(eqc->data[1])
-Int64
-```
-
-This is just a `var`, `value` list of pairs, indicating that `T` now has the value `Int64`.  If
-you now allow `intersect_tuple` to finish and keep progressing, you'll eventually get to `type_intersection_matching`.
- This function contains a call to `solve_tvar_constraints`.  Roughly speaking, `eqc` defines
-`T = Int64`, but `env` defines it as `Int8`; this conflict is detected in `solve_tvar_constraints`
-and the resulting return is `jl_bottom_type`, aka `Union{}`.
 
 ## Subtyping and method sorting
 
-Armed with this knowledge, you may find yourself surprised by the following:
-
-```julia
-julia> typeintersect(Tuple{Array{Int},Float64}, Tuple{Array{T},T})
-Union{}
-
-julia> Tuple{Array{Int},Float64} <: Tuple{Array{T},T}
-true
-```
-
-where `T` is a bound `TypeVar`.  In other words, `A <: B` does not imply that `typeintersect(A, B) == A`.
- A little bit of digging reveals the reason why: `jl_subtype_le` does not use the `cenv_t` constraints
-that we just saw in `typeintersect`.
-
-`jltypes.c` contains three closely related collections of functions for testing how types `a`
-and `b` are ordered:
-
-  * The `subtype` functions implement `a <: b`. Among other uses, they serve in matching function
-    arguments against method signatures in the function cache.
-  * The `type_morespecific` functions are used for imposing a partial order on functions in method
-    tables (from most-to-least specific). Note that `jl_type_morespecific(a,b,0)` really means "is
-    `a` at least as specific as `b`?" and not "is `a` strictly more specific than `b`?"
-  * The `type_match` functions are similar to `type_morespecific`, but additionally accept (and employ)
-    an environment to constrain typevars. The related `type_match_morespecific` functions call `type_match`
-    with an argument `morespecific=1`
-
-All three of these take an argument, `invariant`, which is set to 1 when comparing type parameters
-and otherwise is 0.
+The `type_morespecific` functions are used for imposing a partial order on functions in method
+tables (from most-to-least specific). Note that `jl_type_morespecific(a,b)` really means "is `a`
+at least as specific as `b`?" and not "is `a` strictly more specific than `b`?"
 
-The rules for these are somewhat different. `subtype` is sensitive to the number arguments, but
-`type_morespecific` may not be. In particular, `Tuple{Int,AbstractFloat}` is more specific than
-`Tuple{Integer}`, even though it is not a subtype.  (Of `Tuple{Int,AbstractFloat}` and `Tuple{Integer,Float64}`,
-neither is more specific than the other.)  Likewise, `Tuple{Int,Vararg{Int}}` is not a subtype
-of `Tuple{Integer}`, but it is considered more specific. However, `morespecific` does get a bonus
-for length: in particular, `Tuple{Int,Int}` is more specific than `Tuple{Int,Vararg{Int}}`.
+If `a` is a subtype of `b`, then it is automatically considered more specific.
```

Comment:
I guess this statement means "If `a` is a (non-strict) subtype of `b`, then it is automatically considered at least as specific."? Or does this statement instead mean "If `a` is a (strict) subtype of `b`, then it is automatically considered (strictly) more specific.", or something else?